### ISSUE TITLE
Script not working using any type of midi file

### ISSUE THREAD

**Issue by bjansen99** [Cantersoft/YTPMVE#14]
I cant get it to work, when i run the script it gives me this error message:

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at EntryPoint.FromVegas(Vegas vegas)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at ScriptPortal.Vegas.ScriptHost.ScriptManager.Run(Assembly asm, String className, String methodName)
   at ScriptPortal.Vegas.ScriptHost.RunScript(Boolean fCompileOnly)

EDIT: Some midi files do work but after generating the timeline Vegas just crashes. Does someone have a working midi sample that i can try out?

**Comment by bjansen99** [comment]
Ive tried using your fixed midi file that i found in some other issue thread but now i get a timeline thats 16+ hours long wich is incorrect
![7bd21f5bc1e45d60f65485c515adaf58](https://user-images.githubusercontent.com/96656635/153750544-403a0250-af61-450b-a577-326d947f0141.png)
.

**Comment by Cantersoft** [comment]
May I ask what language your computer is in? Specifically, do you express decimal values with commas or periods? This issue has popped up with multiple people now and I'm beginning to wonder if it's one of those things.

**Comment by bjansen99** [comment]
My Windows is in Dutch. We express decimal values using periods.

**Comment by Cantersoft** [comment]
In that case, I have no idea what the issue may be, but I'm almost certain it's something in the code I can (hopefully) patch.

We can test some extremely simple MIDI files to try to find the issue.

Try these: [MIDItests.zip](https://github.com/Cantersoft/YTPMVE/files/8156154/MIDItests.zip), and tell me if the results look like the following pictures: 

<img width="704" alt="two notes - 0s 2s" src="https://user-images.githubusercontent.com/46493366/156047217-17d674c2-09d8-4f27-a194-c3e1e9acdc3e.png">
<img width="1054" alt="two notes - 0s 2s - overlapping" src="https://user-images.githubusercontent.com/46493366/156047214-00ba5e08-8282-4d1d-8983-4546ff9f2bd3.png">
<img width="863" alt="two notes - 0s 2 12s" src="https://user-images.githubusercontent.com/46493366/156047212-314db500-d7a2-427d-b4ca-7e380e14bc81.png">

If you could send back screenshots of the results that would be helpful.

**Comment by Kippykip** [comment]
Yeah the majority of MIDI files I stick into it has the same error, only 2 of my saved midi files have worked thus far:

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at EntryPoint.FromVegas(Vegas vegas)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at ScriptPortal.Vegas.ScriptHost.ScriptManager.Run(Assembly asm, String className, String methodName)
   at ScriptPortal.Vegas.ScriptHost.RunScript(Boolean fCompileOnly)

**Comment by Kippykip** [comment]
Some example MIDI files that break it:
[Bad_Toys_3D_Title.zip](https://github.com/Cantersoft/YTPMVE/files/8550105/Bad_Toys_3D_Title.zip)
[kirbymidproje.zip](https://github.com/Cantersoft/YTPMVE/files/8550106/kirbymidproje.zip)

**Comment by Kippykip** [comment]
On further inspection it may actually be mido at fault


B:\Program Files\Vegas\VEGAS Pro 14.0\Script Menu\YTPMVE>"B:\Program Files\Vegas\VEGAS Pro 14.0\Script Menu\YTPMVE\YTPMVE.exe"
Running from bundle
Python 3.8.3
Python is present
Opened C:/Users/Kippykip/Music/shrek1_novocals.mid
Traceback (most recent call last):
  File "YTPMVE.py", line 81, in <module>
    for msg in MIDI_file: # Change this so that we don't look at the if statement except the first few times
  File "mido\midifiles\midifiles.py", line 384, in __iter__
  File "mido\midifiles\tracks.py", line 110, in merge_tracks
  File "mido\midifiles\tracks.py", line 68, in _to_abstime
  File "mido\midifiles\meta.py", line 507, in copy
TypeError: __init__() got an unexpected keyword argument 'type'
[11728] Failed to execute script 'YTPMVE' due to unhandled exception!

B:\Program Files\Vegas\VEGAS Pro 14.0\Script Menu\YTPMVE>

**Comment by AlexDecker64** [comment]
did you figure out a way to fix this

**Comment by Cantersoft** [comment]
Hello, I apologize for the extremely lengthy delay.

I have finally fixed this issue, and I got a few others while I was at it. A while ago I removed mido's key signature handler from the code to solve a different issue, but now I can't reproduce that issue-- and it seems the key signature handler's absence was the cause of the error in this case.

[Try out the latest release!](https://github.com/Cantersoft/YTPMVE/releases)
